pointer_test.go

```diff
@@ -0,0 +1,71 @@
+package atomic
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestPointer(t *testing.T) {
+	type foo struct{ v int }
+
+	i := foo{42}
+	j := foo{0}
+	k := foo{1}
+
+	tests := []struct {
+		desc      string
+		newAtomic func() *Pointer[foo]
```

Comment:
this won't compile on Go 1.17, right?

Reply:
oops, the test should also be gated, you're right.